### PR TITLE
website/docs: fix nginx ingress proxy example

### PR DESCRIPTION
**Pull request by eliottness**
## Details

👋 Hey,

I was trying to add an ingress for my k8s cluster to setup a proxy application and copy-pasting the example clearly did not work. Here is a fix for that.

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

File: website/docs/providers/proxy/_nginx_ingress.md [hunk 1]
```diff
@@ -4,20 +4,22 @@ Create a new ingress for the outpost
...
-              backend:
-                  # Or, to use an external Outpost, create an ExternalName service and reference that here.
-                  # See https://kubernetes.io/docs/concepts/services-networking/service/#externalname
-                  service:
-                      name: ak-outpost-example-outpost
-                      port:
-                          number: 9000
+  rules:
+    - host: app.company
+      ingressClassName: nginx
```
Comment (rissson) [inline website/docs/providers/proxy/_nginx_ingress.md:11]:
```suggestion
```

This is very much dependent on the user's kubernetes installation, let's not specify it. Furthermore, it should be `.spec.ingressClassName`, not `.spec.rules.ingressClassName`.

**Review by rissson — CHANGES_REQUESTED** [review]
Hello Elliot, fancy seeing you here! Thank you for your contribution!

**Review by rissson — APPROVED** [review]

**Comment by eliottness** [comment]
I don't have an external outpost but I have this if it can help:
```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: app
  namespace: app
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt
    nginx.ingress.kubernetes.io/auth-url: http://ak-outpost-authentik-embedded-outpost.authentik.svc.cluster.local:9000/outpost.goauthentik.io/auth/nginx
    nginx.ingress.kubernetes.io/auth-signin: https://app.domain.com/outpost.goauthentik.io/start?rd=$escaped_request_uri
    nginx.ingress.kubernetes.io/auth-response-headers: Set-Cookie,X-authentik-username,X-authentik-groups,X-authentik-email,X-authentik-name,X-authentik-uid
    nginx.ingress.kubernetes.io/auth-snippet: |
      proxy_set_header X-Forwarded-Host $http_host;
  labels:
    name: app
spec:
  rules:
  - host: app.domain.com
    http:
      paths:
      - pathType: Prefix
        path: "/outpost.goauthentik.io"
        backend:
          service:
            name: authentik
            port: 
              number: 9000
      - pathType: Prefix
        path: "/"
        backend:
          service:
            name: app
            port: 
              number: 80
  tls:
    - hosts:
      - app.domain.com
      secretName: letsencrypt
```

I had to add `controller.allowSnippetAnnotations=true` to the nginx ingress deployment to register the `auth-snipper` annotation

**Comment by eliottness** [comment]
@drg-tanasescu :point_up:

**Comment by drg-tanasescu** [comment]
Thank you, but for this to work you need to have the app and the authentik in same namespace or am i wrong ? 

`  - host: app.domain.com
    http:
      paths:
      - pathType: Prefix
        path: "/outpost.goauthentik.io"
        backend:
          service:
            name: authentik
            port: 
              number: 9000`

**Comment by drg-tanasescu** [comment]
I tried with a service like this and didn't work. 
```
apiVersion: v1
kind: Service
metadata:
  name: authentik-external-service
spec:
  type: ExternalName
  externalName: auth.dev.example.com
```  

This svc is not created automatically: ak-outpost-authentik-embedded-outpost
```
$ ktools get svc
authentik-postgresql        ClusterIP 
authentik-postgresql-hl    ClusterIP   
authentik-redis-headless  ClusterIP  
authentik-redis-master     ClusterIP  
authentik-server                ClusterIP  
```

--- 
And this worked (with traefik ingress) 
``` 
---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: whoamiv1
  labels:
    name: whoamiv1
spec:
  replicas: 1
  selector:
    matchLabels:
      task: whoamiv1
  template:
    metadata:
      labels:
        task: whoamiv1
    spec:
      containers:
        - name: whoamiv1
          image: traefik/traefikee-webapp-demo:v2
          args:
            - -ascii
            - -name=whoami-traefik
          ports:
            - containerPort: 80
          readinessProbe:
            httpGet:
              path: /ping
              port: 80
            failureThreshold: 1
            initialDelaySeconds: 2
            periodSeconds: 3
            successThreshold: 1
            timeoutSeconds: 2
---
apiVersion: v1
kind: Service
metadata:
  name: whoamiv1
spec:
  ports:
    - name: http
      port: 80
  selector:
    task: whoamiv1

---
apiVersion: traefik.io/v1alpha1
kind: Middleware
metadata:
  name: middlewares-authentik
  namespace: internal-tools
spec:
  forwardAuth:
    address: https://auth.dev.example.com/outpost.goauthentik.io/auth/traefik
    trustForwardHeader: true
    authResponseHeaders:
      - X-authentik-username
      - X-authentik-groups
      - X-authentik-email
      - X-authentik-name
      - X-authentik-uid
      - X-authentik-jwt
      - X-authentik-meta-jwks
      - X-authentik-meta-outpost
      - X-authentik-meta-provider
      - X-authentik-meta-app
      - X-authentik-meta-version
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: whoami-http-ing
  annotations:
    traefik.ingress.kubernetes.io/router.entrypoints: web
    kubernetes.io/ingress.class: traefik-internal
    traefik.ingress.kubernetes.io/router.middlewares: internal-tools-middlewares-authentik@kubernetescrd
spec:
  rules:
    - host: whoami-internal.dev.example.com
      http:
        paths:
          - path: /
            pathType: Prefix
            backend:
              service:
                name: whoamiv1
                port:
                  number: 80

```

I will create tomorrow an issue if i get no answers on the Discord channel and ping you there, hope we can create a working example like this one with traefik.

**Comment by eliottness** [comment]
@drg-tanasescu All the mentions to `ak-outpost-authentik-embedded-outpost.authentik.svc.cluster.local` in my code are referring to the embedded outpost (e.g. authentik itself). It should be changed to the url of your outpost.

**Pull request merged**